drivers/flash/soc_flash_mcux.c

```diff
@@ -87,9 +174,21 @@ static int flash_mcux_read(struct device *dev, off_t offset,
 	 */
 	addr = offset + priv->pflash_block_base;
 
-	memcpy(data, (void *) addr, len);
+#ifdef CONFIG_CHECK_BEFORE_READING
+	rc = is_area_readable(addr, len);
+#endif
 
-	return 0;
+	if (!rc) {
+		memcpy(data, (void *) addr, len);
+#ifdef CONFIG_CHECK_BEFORE_READING
+	} else if (rc == -ENODATA) {
+		/* Erased area, return dummy data as an erased page. */
+		memset(data, 0xFFFFFFFF, len);
```

Comment:
```suggestion
		memset(data, 0xFF, len);
		
```

Reply:
thank you @nvlsianpu i've made the changes you suggested